code/modules/mob/mob.dm

```diff
@@ -1098,23 +1098,25 @@ var/list/slot_equipment_priority = list( \
 				if(!L.key)
 					creatures += L
 		var/picked = input("Please select an NPC to respawn as", "Respawn as NPC")  as null|anything in creatures
-		var/mob/dead/observer/O = src
 		switch(picked)
 			if("Mouse")
-				become_mouse(O)
+				GLOB.respawnable_list -= usr
+				become_mouse()
+				spawn(5)
+					GLOB.respawnable_list += usr
 			else
 				var/mob/living/NPC = picked
 				if(istype(NPC) && !NPC.key)
-					if(mind)
-						O.mind.transfer_to(picked)
-						GLOB.non_respawnable_keys -= ckey
-						O.reenter_corpse()
+					GLOB.respawnable_list -= usr
+					NPC.key = key
+					spawn(5)
```

Comment:
Ditto

Reply:
I think you'll find that that is the original file. I'm not keen on touching it again, for the time being at least.